malsim/sims/mal_simulator.py

```diff
@@ -795,7 +878,11 @@ def step(self, actions):
             "Stepping through iteration %d/%d", self.cur_iter, self.max_iter)
         logger.debug("Performing actions: %s", actions)
 
-        performed_actions = []
+        # Map agent to defense/attack steps performed in this step
+        performed_actions = {}
+
+        # TODO: where do we make sure defender is run before attacker?
+
```

Comment:
Line 487 of the new code in `register_defender`. It used to have a docstring explaining this but you removed it ... irony. :)